libkbfs/folder_block_manager.go

```diff
@@ -61,9 +61,11 @@ type folderBlockManager struct {
 	// MD write. These blocks should be deleted as soon as we know
 	// for sure that the MD write isn't visible to others.
 	// TODO: Persist these to disk?
-	blocksToDeleteChan      chan blocksToDelete
-	blocksToDeletePauseChan chan (<-chan struct{})
-	blocksToDeleteWaitGroup RepeatedWaitGroup
+	blocksToDeleteChan       chan blocksToDelete
+	blocksToDeletePauseChan  chan (<-chan struct{})
+	blocksToDeleteWaitGroup  RepeatedWaitGroup
+	blocksToDeleteCancel     context.CancelFunc
+	blocksToDeleteCancelLock sync.Mutex
```

Comment:
Please put the lock above the thing it protects (with maybe a blank line above it to make things clear).